### PR TITLE
feat: Compose for web in alpha 

### PR DESCRIPTION
**Pull request by AlejandraPedroza**
This PR contains the updates regarding Compose Multiplatform for Web.

### CODE REVIEW

File: images/compose-create-first/multiplatform-web-wizard.png
Comment (eymar) [inline images/compose-create-first/multiplatform-web-wizard.png:1]: I think we'll need to update this screenshot once again after the wizard is updated itself.

Currently "Web" has "Experimental" label. But it will become "Alpha" in 1.6.10
Maybe this screenshot can be already updated by modifying the label in html of the wizard page locally.

**Review by eymar — COMMENTED** [review]
Thank you! :)

File: images/compose-create-first/multiplatform-web-wizard.png
Comment (AlejandraPedroza) [reply images/compose-create-first/multiplatform-web-wizard.png:1]: I changed it locally, please check.

File: images/compose-create-first/modules-structure.png
[the remark this answers is not in the document]
Comment (AlejandraPedroza) [reply images/compose-create-first/modules-structure.png:1]: I think I fixed it, I don't know why it wasn't showing correctly.

[the remark this answers is not in the document]
Comment (eymar) [reply images/compose-create-first/modules-structure.png:1]: Yes, now the page displays a new image!

File: topics/compose-onboard/compose-multiplatform-explore-composables.md [hunk 1]
```diff
@@ -137,6 +138,33 @@ fun main() = application {
...
+//Second variant:
+@OptIn(ExperimentalComposeUiApi::class)
+fun main() {
+    ComposeViewport(viewportContainerId = "composeApplication") { App() }
+}
+```
+
+* In the first variant, the web app is inserted into the container assigned to the `viewportContainer` parameter.
+  The `viewportContainer` parameter uses the entire body of the HTML document as the container for rendering the UI.
+* In the second variant, `composeApplication` is the ID of the container added in your `index.html` file, where the UI is rendered.
```
Comment (zamulla) [inline topics/compose-onboard/compose-multiplatform-explore-composables.md:162]: I'd rephrase the second variant to make the difference clearer: in the first variant, the web app is inserted into X, in the second, the web app is inserted into Y. Maybe "rendered" is more precise, but the sentence structure should be the same for both.

File: topics/compose-onboard/compose-multiplatform-explore-composables.md [hunk 2]
```diff
@@ -137,6 +138,33 @@ fun main() = application {
...
+
+//Second variant:
+@OptIn(ExperimentalComposeUiApi::class)
+fun main() {
+    ComposeViewport(viewportContainerId = "composeApplication") { App() }
+}
+```
+
+* In the first variant, the web app is inserted into the container assigned to the `viewportContainer` parameter.
+  The `viewportContainer` parameter uses the entire body of the HTML document as the container for rendering the UI.
```
Comment (zamulla) [inline topics/compose-onboard/compose-multiplatform-explore-composables.md:161]: I don't know if I would say that a parameter uses something: parameters store things, and it's the job of functions and classes to do things. So it's the `ComposeViewport()` function using the HTML body in this case.

File: topics/compose-onboard/compose-multiplatform-setup.md [hunk 3]
```diff
@@ -57,6 +57,17 @@ We recommend that you install the latest stable versions for compatibility and b
             <p>To update the plugin, on the Android Studio welcome screen, select <strong>Plugins | Installed</strong>. Click <strong>Update</strong> next to Kotlin. You can also check the Kotlin version in <strong>Tools | Kotlin | Configure Kotlin Plugin Updates</strong>.</p>
             <p>The Kotlin plugin should be compatible with the Kotlin Multiplatform Mobile plugin. Refer to the <a href="https://kotlinlang.org/docs/multiplatform-plugin-releases.html#release-details">compatibility table</a>.</p></td>
    </tr>
+    <tr>
+        <td>Browsers</td>
+        <td>
+            <p>To run multiplatform applications in a browser, you need a browser supporting the <a href="https://github.com/WebAssembly/gc">Wasm Garbage Collection (GC) feature</a>.</p>
+            <list>
+                <li><strong>Chrome and Chromium-based:</strong> works by default from version 119 and subsequent versions.</li>
```
Comment (zamulla) [inline topics/compose-onboard/compose-multiplatform-setup.md:65]:
```suggestion
                <li><strong>Chrome and Chromium-based:</strong> works by default starting from version 119.</li>
```

File: topics/compose-onboard/compose-multiplatform-setup.md [hunk 4]
```diff
@@ -57,6 +57,17 @@ We recommend that you install the latest stable versions for compatibility and b
...
+                <li><strong>Firefox:</strong> works by default from version 120 and subsequent versions.</li>
```
Comment (zamulla) [inline topics/compose-onboard/compose-multiplatform-setup.md:66]:
```suggestion
                <li><strong>Firefox:</strong> works by default starting from version 120.</li>
```

File: topics/compose-onboard/compose-multiplatform-setup.md [hunk 5]
```diff
@@ -57,6 +57,17 @@ We recommend that you install the latest stable versions for compatibility and b
...
+                <li><strong>Safari/WebKit:</strong> Wasm GC support is currently under <a href="https://bugs.webkit.org/show_bug.cgi?id=247394">active development.</a></li>
```
Comment (zamulla) [inline topics/compose-onboard/compose-multiplatform-setup.md:67]:
```suggestion
                <li><strong>Safari/WebKit:</strong> Wasm GC support is currently under <a href="https://bugs.webkit.org/show_bug.cgi?id=247394">active development</a>.</li>
```

File: topics/compose-onboard/compose-multiplatform-setup.md [hunk 6]
```diff
@@ -143,6 +154,9 @@ To make sure everything works as expected, install and run the KDoctor tool:
               <li><code>command not found: java</code> — <a href="https://www.oracle.com/java/technologies/javase-downloads.html">install Java</a>.</li>
            </list>
    </def>
+   <def title="Browsers">
+      Make sure that your browser version supports the new <a href="https://github.com/WebAssembly/gc">WasmGC</a> by default. If you are working with older browser versions, you need to <a href="https://kotlinlang.org/docs/wasm-troubleshooting.html#browser-versions">configure the environment</a>.
```
Comment (zamulla) [inline topics/compose-onboard/compose-multiplatform-setup.md:158]:
```suggestion
      Make sure that your browser version supports the new <a href="https://github.com/WebAssembly/gc">WasmGC</a> by default. If you are working with older browser versions, <a href="https://kotlinlang.org/docs/wasm-troubleshooting.html#browser-versions">configure the environment</a>.
```

File: topics/multiplatform-publish-apps.md [hunk 7]
```diff
@@ -71,4 +73,55 @@ file. This helps you analyze crashes that happen in the shared module's code.
 
 When an iOS app is rebuilt from bitcode, its `dSYM` file becomes invalid. For such cases, you can compile the shared module
 to a static framework that stores the debug information inside itself. For instructions on setting up crash report
-symbolication in binaries produced from Kotlin modules, see the [Kotlin/Native documentation](https://kotlinlang.org/docs/native-ios-symbolication.html).
\ No newline at end of file
+symbolication in binaries produced from Kotlin modules, see the [Kotlin/Native documentation](https://kotlinlang.org/docs/native-ios-symbolication.html).
+
+## Web app
+
+To publish your web application, you first need to create the artifacts containing the compiled files
```
Comment (zamulla) [inline topics/multiplatform-publish-apps.md:80]:
```suggestion
To publish your web application, create the artifacts containing the compiled files 
```

File: topics/multiplatform-publish-apps.md [hunk 8]
```diff
@@ -71,4 +73,55 @@ file. This helps you analyze crashes that happen in the shared module's code.
...
+   wasmJsBrowserDistribution
+   ```
+
+4. Click **OK**.
+
+Now, you can use this configuration to run the task:
+
+![Run the Wasm distribution task](compose-run-wasm-distribution-task.png){width=350}
+
+Once the application task completes, you can find the generated artifacts in the `composeApp/build/dist/wasmJs/productionExecutable`
```
Comment (zamulla) [inline topics/multiplatform-publish-apps.md:101]:
```suggestion
Once the task completes, you can find the generated artifacts in the `composeApp/build/dist/wasmJs/productionExecutable`
```

This word seems forgotten here, unless I misuderstood something.

File: topics/multiplatform-publish-apps.md [hunk 9]
```diff
@@ -71,4 +73,55 @@ file. This helps you analyze crashes that happen in the shared module's code.
...
+directory:
+
+![Artifacts directory](compose-web-artifacts.png){width=600}
+
+### Publish your application on GitHub Pages
+
+With the artifacts ready, you can deploy your application on a web hosting platform. In this example, we use GitHub Pages:
```
Comment (zamulla) [inline topics/multiplatform-publish-apps.md:108]:
```suggestion
With the artifacts ready, you can deploy your application on a web hosting platform. In this example, we use GitHub Pages:
```
```suggestion
With the artifacts ready, you can deploy your application on the web hosting platform:
```

We mentioned the platform we're using a couple of times already, might be enough. Tell me what you think.

File: topics/multiplatform-publish-apps.md [hunk 10]
```diff
@@ -71,4 +73,55 @@ file. This helps you analyze crashes that happen in the shared module's code.
...
+
+1. Copy all the contents in your `productionExecutable` directory into the repository where you want to create a site.
```
Comment (zamulla) [inline topics/multiplatform-publish-apps.md:110]:
```suggestion
1. Copy the contents in your `productionExecutable` directory into the repository where you want to create a site.
```

File: topics/multiplatform-publish-apps.md [hunk 11]
```diff
@@ -71,4 +73,55 @@ file. This helps you analyze crashes that happen in the shared module's code.
...
+3. In a browser, navigate to your GitHub pages domain.
+
+   ![Navigate to GitHub pages](publish-your-application-on-web.png){width=650}
+
+   Congratulations! You have published your artifacts on GitHub pages.
+
+### Debug your web application
+
+You can debug your web application in your browser out of the box, without additional configurations. To learn how to debug
+your web application, see the [Debug in your browser](https://kotlinlang.org/docs/wasm-debugging.html#debug-in-your-browser)
```
Comment (zamulla) [inline topics/multiplatform-publish-apps.md:126]:
```suggestion
in the browser, see the [Debug in your browser](https://kotlinlang.org/docs/wasm-debugging.html#debug-in-your-browser)
```

**Review by zamulla — CHANGES_REQUESTED** [review]
Thanks! Left several suggestions to consider.

File: topics/compose-onboard/compose-multiplatform-explore-composables.md [hunk 12]
```diff
@@ -137,6 +138,33 @@ fun main() = application {
 Currently, the `App` function doesn't declare any parameters. In a larger application, you typically pass parameters to
 platform-specific dependencies. These dependencies could be created by hand or using a dependency injection library.
 
+### On the web
```
[the remark this answers is not in the document]
Comment (AlejandraPedroza) [reply topics/compose-onboard/compose-multiplatform-explore-composables.md:141]: Agree!

File: topics/compose-onboard/compose-multiplatform-explore-composables.md (hunk 1 above)
Comment (AlejandraPedroza) [reply topics/compose-onboard/compose-multiplatform-explore-composables.md:162]: I rephrased it. What do you think?

File: topics/multiplatform-publish-apps.md [hunk 13]
```diff
@@ -71,4 +73,55 @@ file. This helps you analyze crashes that happen in the shared module's code.
...
+Once the task completes, you can find the generated artifacts in the `composeApp/build/dist/wasmJs/productionExecutable`
+directory:
+
+![Artifacts directory](compose-web-artifacts.png){width=600}
+
+### Publish your application on GitHub Pages
+
+With the artifacts ready, you can deploy your application on the web hosting platform:
+
+1. Copy the contents in your `productionExecutable` directory into the repository where you want to create a site.
```
Comment (zamulla) [inline topics/multiplatform-publish-apps.md:110]:
```suggestion
1. Copy the contents of your `productionExecutable` directory into the repository where you want to create a site.
```

I thought something else was off in this phrase.

**Review by zamulla — APPROVED** [review]
Otherwise all good, thank you!
